app/Adapters/MagentoAdapter.php

```diff
@@ -323,4 +331,26 @@ public function findProductsBySku(array|Collection $skus, int $perPage = 50): La
             new Condition('status', 'eq', 1),
         ));
     }
+
+    public function getLabel(string $labelKey): array
+    {
+        $cacheId = fn (string $key) => __METHOD__.":{$key}";
+
+        $existingEntry = cache()->get($cacheId($labelKey));
```

Comment:
:large_blue_circle: I think here it would be better to use `has()` instead. It makes the code a little bit more consistent and avoids the double retrieval of the same cache value.

```suggestion
        $existingEntry = cache()->has($cacheId($labelKey));
```